src/core/ack_tracker.c

```diff
@@ -96,6 +96,39 @@ QuicAckTrackerAddPacketNumber(
         !RangeUpdated;
 }
 
+BOOLEAN
+QuicAckTrackerDidHitReorderingThreshold(
```

Comment:
Let's put a comment above this that indicated this implements the logic defined in https://datatracker.ietf.org/doc/html/draft-ietf-quic-ack-frequency#name-response-to-out-of-order-pa.